app/cryptography/token_helper.py

```diff
@@ -0,0 +1,97 @@
+import json
+
+from jwcrypto import jwe, jwt
+from jwcrypto.common import base64url_decode
+from jwcrypto.jwe import InvalidJWEData
+from jwcrypto.jws import InvalidJWSSignature, InvalidJWSObject
+from jwcrypto.jwt import JWTInvalidClaimFormat, JWTMissingClaim, JWTExpired
+
+from app.authentication.invalid_token_exception import InvalidTokenException
+
+
+def decrypt_jwe(encrypted_token, secret_store, purpose, default_kid=None):
+    try:
+        jwe_token = jwe.JWE(algs=['RSA-OAEP', 'A256GCM'])
+        jwe_token.deserialize(encrypted_token)
+
+        jwe_kid = extract_kid_from_header(encrypted_token, default_kid)
+
+        sr_private_jwk = secret_store.get_private_key_by_kid(purpose, jwe_kid)
+
+        jwe_token.decrypt(sr_private_jwk)
+
+        return jwe_token.payload.decode()
+    except InvalidJWEData as e:
+        raise InvalidTokenException(repr(e))
+
+
+def encrypt_jwe(payload, kid, secret_store, purpose, alg="RSA-OAEP", enc="A256GCM", protected_header=None):
+
+    try:
+        downstream_public_jwk = secret_store.get_public_key_by_kid(purpose, kid)
+
+        _protected_header = {"alg": alg, "enc": enc, "kid": kid} if protected_header is None else protected_header
+
+        token = jwe.JWE(plaintext=payload, protected=_protected_header)
+
+        token.add_recipient(downstream_public_jwk)
+
+        return token.serialize(compact=True)
+    except InvalidJWEData as e:
+        raise InvalidTokenException(repr(e))
+
+
+def decode_jwt(jwt_token, secret_store, purpose, check_claims=None):
+    try:
+        jwt_kid = extract_kid_from_header(jwt_token)
+
+        upstream_public_jwk = secret_store.get_public_key_by_kid(purpose, jwt_kid)
+
+        _check_claims = {"exp": None, "iat": None} if check_claims is None else check_claims
```

Comment:
It looks like check_claims is only overridable for the tests, I think it shouldn't be and we should re-work the tests so it's not needed.